Update Prow Jobs documentation (#5479)
kyma-project/test-infra: docs/prow/prow-jobs.md

`/retest` to only rerun failed tests
`/test {test-name}` or `/retest {test-name}` to only rerun a specific test. For example, run `/test pre-main-kyma-components-binding-usage-controller`.

After you trigger the job, it appears on [`https://status.build.kyma-project.io/`](https://status.build.kyma-project.io/).

## Create jobs

> Jobs are generated from templates stored in the `templates` directory. For details on how to configure templates, see [this](./manage-component-jobs-with-templates.md) document. To generate jobs, run one of these commands in the root of the repository:

Reviewer: ```suggestion
Jobs are generated from templates stored in the `templates` directory. For details on how to configure templates, read [Manage component jobs with templates](./manage-component-jobs-with-templates.md). To generate jobs, run one of these commands in the root of the repository:
```

```bash
go run development/tools/cmd/rendertemplates/main.go --data path/to/directory/with/data/files
```
or
```bash
make jobs-definitions
```

- For details on how to create jobs, see [Manage component jobs with templates](./manage-component-jobs-with-templates.md).
- For further reference, read a more technical insight into the Kubernetes [Prow Jobs](https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md).

## Prow job tester
Prow job tester (`pjtester`) is a dedicated tool for testing changes to Prow jobs and scripts in the `test-infra` repository,

> which are **under development**. Additionally, it can use code from pull requests (PRs) in other repositories.

Reviewer: ```suggestion
```

For example, to test PR in the Kyma repository create a new file `vpath/pjtester.yaml` in the `test-infra` repository.
```yaml
pjNames:
  - pjName: <PROW JOB NAME>
  - pjName: ...
prConfigs:
  kyma-project:
    kyma:
      prNumber: <PR NUMBER IN KYMA REPO> 
```
| Parameter name | Required | Description |
|----------------|----------|-------------|
| **pjNames** | Yes | List containing the configuration of Prow jobs to test.
| **pjNames.pjName** | Yes | Name of the Prow job to test.
| **prConfigs** | No | Dictionary containing the numbers of the pull request on repositories other than `test-infra`. <br> `pjtester` uses their code to test the Prow jobs.

> > **NOTE:** It is recommended to keep PRs as draft ones until you're satisfied with the results.

Reviewer: ```suggestion
> **NOTE:** We recommend to keep PRs as draft ones until you're satisfied with the results.
```

> - For more details on how to use `pjtester`, see [this](https://github.com/kyma-project/test-infra/blob/main/development/tools/cmd/pjtester/README.md)

Reviewer: ```suggestion
- For more details on how to use `pjtester`, see [the `pjtester` README](https://github.com/kyma-project/test-infra/blob/main/development/tools/cmd/pjtester/README.md)
```
